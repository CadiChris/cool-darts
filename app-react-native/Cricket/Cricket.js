import React from 'react'
import TableauDesScores from './TableauDesScores/TableauDesScores'
import { StyleSheet, Text, View } from 'react-native'

import Partie from './Partie/Partie'

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Partie style={{height: '40%'}}/>
        <TableauDesScores />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#aaa',
    paddingTop: 25
  },
});
