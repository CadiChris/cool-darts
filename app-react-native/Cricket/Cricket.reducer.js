import { scoreVierge } from './TableauDesScores/arbitrage/score'
import { calculerLeNouveauScore } from './TableauDesScores/arbitrage/arbitre'
import { vainqueurs } from './TableauDesScores/arbitrage/vainqueurs'

const STATE_INITIAL = {
  phase: 'INSCRIPTION',
  joueurs: [],
  peutDemarrer: false,
  scores: [],
  vainqueurs: []
}

export default function partie(state = STATE_INITIAL, action) {
  switch (action.type) {
    case 'INSCRIRE_JOUEUR':
      return {
        ...state,
        joueurs: [...state.joueurs, action.joueur],
        scores: [...state.scores, scoreVierge(action.joueur)],
        peutDemarrer: state.joueurs.length > 0
      }

    case 'DEMARRER_PARTIE':
      return {
        ...state,
        phase: 'EN_COURS'
      }

    case 'LANCER_FLECHETTE':
      const lancer = {
        lanceur: action.joueur,
        chiffre: action.chiffre,
        touches: action.touches
      }
      const nouveauScore = calculerLeNouveauScore(state.scores, lancer)
      const vainqueursDuNouveauScore = vainqueurs(nouveauScore)

      return {
        ...state,
        scores: nouveauScore,
        vainqueurs: vainqueursDuNouveauScore,
        phase: vainqueursDuNouveauScore.length > 0 ? 'TERMINEE' : state.phase
      }


    default:
      return state
  }
}