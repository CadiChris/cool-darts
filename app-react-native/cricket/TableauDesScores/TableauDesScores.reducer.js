import { scoreVierge } from './arbitrage/score'
import { calculerLeNouveauScore } from './arbitrage/arbitre'

const stateInitial = {
  scores: []
}

export default function tableauDesScores(tableauDesScores = stateInitial, action) {
  switch (action.type) {

    case 'INSCRIRE_JOUEUR' :
      return {
        scores: [
          ...tableauDesScores.scores,
          scoreVierge(action.joueur)
        ]
      }

    case 'LANCER_FLECHETTE' :
      const lancer = {
        lanceur: action.joueur,
        chiffre: action.chiffre,
        touches: action.touches
      }
      return {
        scores: calculerLeNouveauScore(tableauDesScores.scores, lancer)
      }

    default:
      return tableauDesScores
  }
}