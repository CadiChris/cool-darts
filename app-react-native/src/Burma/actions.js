import Contrats from "./arbitrage/Contrats";

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = joueurs => ({
  type: DEMARRER_PARTIE,
  joueurs
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (lanceur, nombreDeBulls) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeBulls)
  }
});
