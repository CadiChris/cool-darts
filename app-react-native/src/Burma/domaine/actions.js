import Contrats from './arbitrage/Contrats';

export const DEMARRER_BURMA = 'BURMA/DEMARRER';
export const demarrerBurma = joueurs => ({
  type: DEMARRER_BURMA,
  joueurs,
});

export const VOLEE = 'BURMA/VOLEE';
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches),
  },
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches),
  },
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches),
  },
});
