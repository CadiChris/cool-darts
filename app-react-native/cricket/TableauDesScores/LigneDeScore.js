import React from 'react'
import { connect } from 'react-redux'
import { Text, View, TouchableHighlight } from 'react-native'
import { Col, Row } from "react-native-easy-grid"
import { Styles, Tailles } from './Styles'
import { lancerFlechette } from './TableauDesScores.actions'

class LigneDeScore extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const {score} = this.props
    return (
      <Row style={Styles.bordureBasse}>
        <Col style={{justifyContent: 'center'}} size={Tailles.largeurDeLaColonneJoueur}>
          <Text style={{paddingLeft: 10}}>{score.joueur} - {score.points}</Text>
        </Col>

        {
          Object.keys(score.cible).map((chiffre, index) => {
            const {touches, ferme} = score.cible[chiffre]
            return (
              <Col style={[{flex:1, flexDirection:'row', alignItems:'stretch'}, Styles.bordureDroite, ferme ? Styles.chiffreFerme : {}]} key={index}>
                <TouchableHighlight onPress={() => this.lancerSimpleDans(chiffre)} style={{alignSelf: 'stretch'}}>
                  <View>
                    <Text> </Text>
                    <Text style={{textAlign: 'center'}}>{'X'.repeat(touches)}</Text>
                  </View>
                </TouchableHighlight>
              </Col>
            )
          })
        }
      </Row>
    )
  }

  lancerSimpleDans(chiffre) {
    const {dispatch, score} = this.props
    dispatch(lancerFlechette(score.joueur, chiffre, 1))
  }
}

export default connect()(LigneDeScore)