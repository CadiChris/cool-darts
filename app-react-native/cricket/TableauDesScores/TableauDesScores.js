import React from 'react'
import { connect } from 'react-redux'
import { StyleSheet, Text, View } from 'react-native'
import { Grid } from "react-native-easy-grid"
import EnTete from './EnTete'
import LigneDeScore from './LigneDeScore'


class TableauDesScores extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const {scores} = this.props
    return (
      <View style={{flex: 1}}>
        <Grid style={{flexDirection: 'column'}}>
          <EnTete />
          {
            scores.map((score, index) => <LigneDeScore score={score} key={index}/>)
          }
        </Grid>
      </View>
    )
  }
}

function mapStateToProps(state) {
  return {
    scores: [...state.tableauDesScores.scores]
  }
}

export default connect(mapStateToProps)(TableauDesScores)
