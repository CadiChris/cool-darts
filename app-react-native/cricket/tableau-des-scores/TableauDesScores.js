import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import EnTete from './EnTete'
import LigneDeScore from './LigneDeScore'

import { Grid, Row } from "react-native-easy-grid";

export default () => {
  return (
    <View style={{flex:1, alignItems:'stretch'}}>
      <Grid>
        <Row style={{height: 45}}><EnTete /></Row>
        <Row><LigneDeScore /></Row>
      </Grid>
    </View>
  )
}