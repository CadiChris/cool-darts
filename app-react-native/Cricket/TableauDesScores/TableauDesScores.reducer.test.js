import tableauDesScores from './TableauDesScores.reducer'
import { lancerFlechette } from './TableauDesScores.actions'
import { inscrireJoueur } from '../Cricket.actions'

describe('reducer', () => {
  it('retourne le state initial', () => {
    expect(tableauDesScores(undefined, {})).toMatchSnapshot()
  })

  it('crée un score vierge pour un nouvel inscrit', () => {
    expect(tableauDesScores(undefined, inscrireJoueur('J1'))).toMatchSnapshot()
  })

  it('accepte un lancer de fléchette', () => {
    const tableauDeTest = tableauDesScores(undefined, inscrireJoueur('J1'))

    expect(
      tableauDesScores(tableauDeTest, lancerFlechette('J1', 20, 1))
    ).toMatchSnapshot()
  })

  it('détecte les vainqueurs', () => {
    const toutFermeSaufLeBull = [15, 16, 17, 18, 19, 20].reduce(
      (tableau, chiffre) => (
        tableauDesScores(tableau, lancerFlechette('J1', chiffre, 3))
      ),
      tableauDesScores(undefined, inscrireJoueur('J1')))

    const fermerLeBull = lancerFlechette('J1', 25, 3)
    expect(tableauDesScores(toutFermeSaufLeBull, fermerLeBull).vainqueurs).toEqual(['J1'])
  })
})
