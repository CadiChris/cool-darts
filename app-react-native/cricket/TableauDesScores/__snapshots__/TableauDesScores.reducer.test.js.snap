// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer accepte un lancer de fléchette 1`] = `
Object {
  "scores": Array [
    Object {
      "cible": Object {
        "15": Object {
          "ferme": false,
          "touches": 0,
        },
        "16": Object {
          "ferme": false,
          "touches": 0,
        },
        "17": Object {
          "ferme": false,
          "touches": 0,
        },
        "18": Object {
          "ferme": false,
          "touches": 0,
        },
        "19": Object {
          "ferme": false,
          "touches": 0,
        },
        "20": Object {
          "ferme": false,
          "touches": 1,
        },
        "25": Object {
          "ferme": false,
          "touches": 0,
        },
      },
      "joueur": "J1",
      "points": 0,
    },
  ],
}
`;

exports[`reducer crée un score vierge pour un nouvel inscrit 1`] = `
Object {
  "scores": Array [
    Object {
      "cible": Object {
        "15": Object {
          "ferme": false,
          "touches": 0,
        },
        "16": Object {
          "ferme": false,
          "touches": 0,
        },
        "17": Object {
          "ferme": false,
          "touches": 0,
        },
        "18": Object {
          "ferme": false,
          "touches": 0,
        },
        "19": Object {
          "ferme": false,
          "touches": 0,
        },
        "20": Object {
          "ferme": false,
          "touches": 0,
        },
        "25": Object {
          "ferme": false,
          "touches": 0,
        },
      },
      "joueur": "J1",
      "points": 0,
    },
  ],
}
`;

exports[`reducer retourne le state initial 1`] = `
Object {
  "scores": Array [],
}
`;
